ui/apps/platform/src/Containers/Collections/CollectionResults.tsx

```diff
@@ -1,7 +1,153 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState, useMemo } from 'react';
+import {
+    Button,
+    Divider,
+    EmptyState,
+    EmptyStateIcon,
+    EmptyStateVariant,
+    Flex,
+    FlexItem,
+    Select,
+    SearchInput,
+    SelectOption,
+    debounce,
+} from '@patternfly/react-core';
+import { ListIcon } from '@patternfly/react-icons';
+import useSelectToggle from 'hooks/patternfly/useSelectToggle';
+import ResourceIcon from 'Components/PatternFly/ResourceIcon';
 
-function CollectionResults() {
-    return <></>;
+import { dryRunCollection } from 'services/CollectionsService';
+import { ListDeployment } from 'types/deployment.proto';
+
+function CollectionResults({ dryRunConfig }) {
+    const { isOpen, onToggle, closeSelect } = useSelectToggle();
+    const [selected, setSelected] = useState('Deployment');
+    const [filterText, setFilterText] = useState('');
+    const [isEndOfResults, setIsEndOfResults] = useState(false);
+    const [deployments, setDeployments] = useState<ListDeployment[]>([]);
+    const [hasRuleSelected, setHasRuleSelected] = useState(false);
+
+    function onRuleOptionSelect(_, value) {
+        setSelected(value);
+        setFilterText('');
+        closeSelect();
+    }
+
+    const fetchDryRun = useCallback(
+        (page: number) => {
+            const pageSize = 10;
+            const query = filterText ? `${selected}:${filterText}` : '';
+            const { request } = dryRunCollection(dryRunConfig, query, page, pageSize);
+            request
+                .then((results) => {
+                    setIsEndOfResults(results.length < 10);
+                    setDeployments((current) => (page === 0 ? results : [...current, ...results]));
+                    setHasRuleSelected(true);
+                })
+                .catch(() => {
+                    // TODO: indicate results not loading properly?
```

Comment:
I think the error handling here is going to be combined with the current "onSave" error handling, since the possible errors that are returned from the dry run API should match what is returned when the user tries to save.

Lets create a new ticket for it though so that it doesn't hold up this PR.

Reply:
https://issues.redhat.com/browse/ROX-13780

Reply:
Does error handling need to distinguish failure of the request itself versus backend validation error?

We might compare to policy dry run, although not assuming they have identical behavior.